var/spack/repos/builtin/packages/py-notebook/package.py

```diff
@@ -12,6 +12,8 @@ class PyNotebook(PythonPackage):
     homepage = "https://github.com/jupyter/notebook"
     pypi = "notebook/notebook-6.1.4.tar.gz"
 
+    version("6.5.7", sha256="04eb9011dfac634fbd4442adaf0a8c27cd26beef831fe1d19faf930c327768e4")
```

Comment:
While there are `7.x` versions available, I didn't want to take on more than was manageable here.